File: maxrecursive.cpp
#include <iostream>

using namespace std;

int maxx(int arr[], int n, int i, int max)
{
    int l = n;
    if(i<l)
    {
        if(arr[i]>max)
        {
          
            return maxx(arr, n, i+1, arr[i]);
        }
        else {
            return maxx(arr, n, i+1, max);
        }
    }
    else {
        return max;
    }
}

int main()
{
    cout<<"Enter the number of elements:";
    int n;cin>>n;
    int arr[n];
    for(int i=1;i<=n;i++)
    {
        cin>>arr[i-1];
    }
    cout<<maxx(arr,n,0,arr[0]);
    
}

File: Maxrecursive.java
import java.util.*;
public class Maxrecursive {

    public static int maxx(int arr[],int i,int max)
    {
        if(i<arr.length)
        {
            if(arr[i]>max)
            {
                return maxx(arr,i+1,arr[i]);
            }
            else{
                return maxx(arr,i+1,max);
            }
        }
        else{
            return max;
        }
    }

    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        System.out.printf("The Number od input :");
        int n=sc.nextInt();
        int arr[]=new int[n];
        for(int i=1;i<=n;i++)
        {
            arr[i-1]=sc.nextInt();
        }

        System.out.println("The maxmum is "+maxx(arr,0,arr[0]));

        sc.close();

    }
    
}
